### ISSUE TITLE
nth not supported on this type: ExceptionInfo

### ISSUE THREAD

**Issue by pupeno** [ptaoussanis/carmine#198]
I'm getting this exception every now and then:

```
17-07-15 06:12:45 machine ERROR [tech.dashman.redis:?] - Listener handler exception
                              java.lang.Thread.run              Thread.java:  745
java.util.concurrent.ThreadPoolExecutor$Worker.run  ThreadPoolExecutor.java:  617
 java.util.concurrent.ThreadPoolExecutor.runWorker  ThreadPoolExecutor.java: 1142
               java.util.concurrent.FutureTask.run          FutureTask.java:  266
                                               ...                               
               clojure.core/binding-conveyor-fn/fn                 core.clj: 1938
                                               ...                               
                         clojure.core/bound-fn*/fn                 core.clj: 1911
                                clojure.core/apply                 core.clj:  650
                                               ...                               
                       clojure.core/with-bindings*                 core.clj: 1881 (repeats 2 times)
                                clojure.core/apply                 core.clj:  646
                                               ...                               
             tech.dashman.redis/eval31714/fn/fn/fn                redis.clj:   33
          tech.dashman.redis/eval31714/fn/fn/fn/fn                redis.clj:   33
             tech.dashman.redis/eval31714/fn/fn/fn                redis.clj:   33
                                               ...                               
java.lang.UnsupportedOperationException: nth not supported on this type: ExceptionInfo
```

That line, redis.clj:33, contains this:

```clojure
(car/with-new-pubsub-listener (:spec @server1-conn) {})
```

I tried wrapping it in a try/catch, to get more information about the exception, but I'm not getting the exception. I don't yet understand why.

**Comment by pupeno** [comment]
I guess this is where the exception is coming from:

https://github.com/ptaoussanis/carmine/blob/10da07f5555cebe54754e5d50d8f12e4dd400193/src/taoensso/carmine.clj#L360

**Comment by ptaoussanis** [comment]
Hi there, could you please confirm what version of Carmine you're using?

This sounds like it it could be due to https://github.com/ptaoussanis/carmine/commit/a997a41e25b262cab3037694560f285c29160caf, fixed in v2.16.0.

Thanks!
